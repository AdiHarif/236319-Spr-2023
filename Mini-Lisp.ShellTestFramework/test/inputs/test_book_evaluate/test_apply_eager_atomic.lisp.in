; =================================================
; apply-eager-atomic - 1 argument functions:
; =================================================
(apply-eager-atomic 'atom '(a) '(t.t nil.nil))
(apply-eager-atomic 'car '((a b)) '(t.t nil.nil))
(apply-eager-atomic 'cdr '((a b)) '(t.t nil.nil))
(apply-eager-atomic 'eval '(a) '(a.x t.t nil.nil))

; =================================================
; apply-eager-atomic - 2 argument functions:
; =================================================
(apply-eager-atomic 'cons '(a b) '(t.t nil.nil))
(apply-eager-atomic 'eq '(a a) '(t.t nil.nil))
(apply-eager-atomic 'eq '(a b) '(t.t nil.nil))
(apply-eager-atomic 'set '(a b) '(t.t nil.nil))
(apply-eager-atomic 'set '(a (b c)) '(t.t nil.nil))

; =================================================
; apply-eager-atomic - the error function:
; =================================================
(apply-eager-atomic 'error NIL '(t.t nil.nil)) ; NIL
(apply-eager-atomic 'error 'my_error '(t.t nil.nil)) ; MY_ERR

; =================================================
; apply-eager-atomic - Errors:
; =================================================
(apply-eager-atomic 'bla '(a b) '(t.t nil.nil))
