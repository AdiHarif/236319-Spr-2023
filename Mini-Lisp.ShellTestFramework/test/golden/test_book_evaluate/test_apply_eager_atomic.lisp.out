T
A
(B)
(A.B)
T
NIL
A
Error [MISSING] in (ATOMIC ACTUALS A-LIST)
Error [A] in (ERROR ACTUALS)
